Refactor top level, part 1

mvahowe opened the pull request:
This PR refactors the metadata.schema.json file to contain just the top-level skeleton, with all the details of sections in separate files. This paves the way for a schema that chooses between multiple top-level skeletons, while maximising usability of the section details.

rdb commented:
This is fine, I didn't think that creating schemas for these arrays was necessary but I don't really mind it either.

Two things to note:
* The removal of index.js means that the supplied ajv-based validator no longer works.  I don't mind if we want to remove that validator, or move whatever is in index.js into the validator instead, but we should do one or the other.
* Files should by convention end with a newline.

mvahowe commented:
@rdb I just replaced the index.js file. You'll notice that the schemas bit at the top is a little different (because, in the store, I'm using this for other schema). It would be easier to maintain this in future if you could change your code to use my format. (I just deleted the 2 mentions of "config".)

The thinking behind moving the arrays into their own files is that, in my next PR, we end up with multiple, slightly different copies of the top-level skeleton, so it makes sense to keep that skeleton as sparse as possible.
